src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -204,12 +213,102 @@ public InnerClassLambdaMetafactory(MethodHandles.Lookup caller,
     }
 
     private static String lambdaClassName(Class<?> targetClass) {
+        return createNameFromTargetClass(targetClass) + counter.incrementAndGet();
+    }
+
+    private static String createNameFromTargetClass(Class<?> targetClass) {
         String name = targetClass.getName();
         if (targetClass.isHidden()) {
             // use the original class name
             name = name.replace('/', '_');
         }
-        return name.replace('.', '/') + "$$Lambda$" + counter.incrementAndGet();
+        return name.replace('.', '/') + "$$Lambda$";
+    }
+
+    /**
+     * Calculate hash value of the given String in the same manner as the
+     * @see java.lang.StringUTF16#hashCode does with difference that hash value
+     * is long instead of int.
+     *
+     * @param name String for which method calculates hash value for
+     *
+     * @return a hash value for the given String
+     * */
+    private String fixedSizeStringHash(String name) {
+        char appendingCharacter = 'a';
+        long h = 0;
+        int length = name.length();
+        for (int i = 0; i < length; i++) {
+            h = 31 * h + name.charAt(i);
+        }
+
+        StringBuilder stable = new StringBuilder(Long.toString(Math.abs(h), Character.MAX_RADIX));
+        int stableLength = stable.length();
+
+        // We need to pad stable lambda name to the fixed size, so we append predefined character to the end of the stable name if needed.
+        // This does not affect stability of the name, just it's length. We want all the hashed names to be of the same length.
+        while (stableLength != targetLengthForStableName) {
+            stable.append(appendingCharacter);
```

Comment:
paddingCharacter?

Reply:
Shouldn't this be a constant?

Reply:
> paddingCharacter?

Changed as suggested.

Reply:
> Shouldn't this be a constant?

I just made this variable `final`. Maybe a better solution is to remove it as a class field `private static final char paddingCharacter = 'a'`?